File: Tests/sample-package/Builder.rb
require_relative "../Scripts/ProjectBuilder.rb"

# See also:
# - ios - How to open a .swiftmodule file - Stack Overflow: http://bit.ly/2HET1d9

class Builder < ProjectBuilder

   def initialize(arch = Arch.default)
      @root = File.dirname(__FILE__)
      super("hello-package", arch)
   end

   def executeBuild
      binary1 = "#{@builds}/#{@config}/Exe"
      binary2 = "#{@builds}/#{@config}/libLib.so"
      execute "cd #{@builds} && #{@swiftBuildCmd}"
      execute "file #{binary1}"
      execute "file #{binary2}"
   end

end

File: Tests/sample-package/Builder.js
const ProjectBuilder = require("../ProjectBuilder")

module.exports = class Builder extends ProjectBuilder {

  constructor(component, arch) {
    super(component, arch)
    this.binary = `${this.buildPath}/release/Exe`
    this.libFilePath = `${this.buildPath}/release/libLib.so`
  }

  executeBuild() {
    this.execute(`cd "${this.rootPath}" && ${this.swiftBuildCmd}`);
    this.execute(`cd "${this.rootPath}" && ${this.swiftBuildCmd} -c release`);

    // Swift Libs
    this.copyLibs()

    this.execute(`file ${this.binary}`)
    this.execute(`file ${this.libFilePath}`)
  }

  get libs() {
    var value = super.libs
    value.push(this.libFilePath)
    return value
  }
};
